README.md

```diff
@@ -1,24 +1,107 @@
+})
+```
 
-#### Project structure
-There are two main folders in the project:
+`Pair` is an abstract class that defines the interface for trading pairs. If you would like to add custom trading functionality, you can extend this class and implement it's methods:
+- `estimateAtoB`: Indicates how to estimate the B Asset the user will receive for the amount of A Asset. For example Fees could be considered here.
+- `estimateBtoA`: Indicates how to estimate the A Asset the user will receive for the amount of B Asset. For example Fees could be considered here.
+- `exchangeAtoB`: Indicates how to send A Asset to be exchanged for B Asset. Here you can add logic to detect exchange finalization for example for a more complex exchange operation like bridges.
+- `exchangeBtoA`: Indicates how to send B Asset to be exchanged for A Asset. Here you can add logic to detect exchange finalization for example for a more complex exchange operation like bridges.
+
+### Project structure
+There are three main folders in the project:
 - `wallet` - is a burner wallet instance ready to use your plugin
 - `test-wallet` - is a burner wallet instance ready to test your plugin and resources in testnets
 - `my-plugin` - is the folder where the plugin code is located. To change the name of the plugin it is necessary to update the folder name `my-plugin` and all mentions to `my-plugin` to the new name of your plugin.
 
 Inside `my-plugin` you can find the files that defines the resources to be exposed by the plugin to be used by the burner wallet in order to interact with the ERC677 to ERC677 bridge extension:
-- `Stake` - extends from `ERC677Asset` defined in `tokenbridge-plugin`
-- `xStake` - extends from `ERC677Asset` defined in `tokenbridge-plugin`
-- `StakeBridge` - extends from `Mediator` defined in `tokenbridge-plugin`
+- `Stake` - extends from `ERC677Asset` defined in `@poanet/tokenbridge-bw-exchange`
+- `xStake` - extends from `ERC677Asset` defined in `@poanet/tokenbridge-bw-exchange`
+- `StakeBridge` - extends from `Mediator` defined in `@poanet/tokenbridge-bw-exchange`
+
+The class `ERC677Asset` extends from `ERC20Asset` and overrides some methods:
+- `_send`: In this case the method `transferAndCall` is used instead of the `transfer` method.
+- `startWatchingAddress`: Add logic to correctly track the transfer events related to the wallet account.
+- `getTx`: Overrides the logic to return the stored information about a transfer transaction to avoid errors on the information displayed.
+
+The class `Mediator` extend from some other classes that has `Pair` as base. It implements the previous mentioned methods to estimate and exchange the assets. It also implements some methods to detect the exchange finalization.
+Here are some details of the implemented methods:
+- `estimateAtoB`: Get the fee and calculates the amount of B Asset the user will receive.
+- `estimateBtoA`: Get the fee and calculates the amount of A Asset the user will receive.
```

Comment:
We need to specify somewhere that there is a special field that can be used to notify an user about exchange properties (we use this field to notify about the transaction fees).